platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java

```diff
@@ -531,7 +535,16 @@ public class SwirldsPlatform implements Platform, Startable {
         final List<Predicate<EventDescriptor>> isDuplicateChecks = new ArrayList<>();
         isDuplicateChecks.add(d -> shadowGraph.isHashInGraph(d.getHash()));
 
-        eventLinker = buildEventLinker(time, isDuplicateChecks);
+        final SyncConfig syncConfig = platformContext.getConfiguration().getConfigData(SyncConfig.class);
+
+        final IntakePipelineManager intakePipelineManager;
+        if (syncConfig.waitForEventsInIntake()) {
+            intakePipelineManager = new DefaultIntakePipelineManager(currentAddressBook);
+        } else {
+            intakePipelineManager = new NoOpIntakePipelineManager();
+        }
```

Comment:
Would it be possible to construct this inside the gossip implementation? Unless we plan on using this for chatter, this is mostly just sync specific. I've not given this a lot of thought... if this turns out to be prohibitively difficult perhaps we can discuss.

Reply:
This was possible after doing the refactor suggested below. Done